lgc/state/Compiler.cpp

```diff
@@ -145,31 +145,26 @@ void PipelineState::generate(std::unique_ptr<Module> pipelineModule, raw_pwrite_
   Timer *codeGenTimer = timers.size() >= 3 ? timers[2] : nullptr;
 
   // Set up "whole pipeline" passes, where we have a single module representing the whole pipeline.
-  //
-  // TODO: The "whole pipeline" passes are supposed to include code generation passes. However, there is a CTS issue.
-  // In the case "dEQP-VK.spirv_assembly.instruction.graphics.16bit_storage.struct_mixed_types.uniform_geom", GS gets
-  // unrolled to such a size that backend compilation takes too long. Thus, we put code generation in its own pass
-  // manager.
-  std::unique_ptr<PassManager> patchPassMgr(PassManager::Create());
-  patchPassMgr->setPassIndex(&passIndex);
-  patchPassMgr->add(createTargetTransformInfoWrapperPass(getLgcContext()->getTargetMachine()->getTargetIRAnalysis()));
+  std::unique_ptr<PassManager> passMgr(PassManager::Create());
+  passMgr->setPassIndex(&passIndex);
+  passMgr->add(createTargetTransformInfoWrapperPass(getLgcContext()->getTargetMachine()->getTargetIRAnalysis()));
 
   // Manually add a target-aware TLI pass, so optimizations do not think that we have library functions.
-  getLgcContext()->preparePassManager(&*patchPassMgr);
+  getLgcContext()->preparePassManager(&*passMgr);
```

Comment:
Ditto

Reply:
If I remove this line, llpc lit tests start failing:
    LLPC_SHADERTEST :: shaderdb/OpExtInst_TestPow2_lit.frag
    LLPC_SHADERTEST :: shaderdb/OpExtInst_TestTanVec4Const_lit.frag
    LLPC_SHADERTEST :: shaderdb/OpExtInst_TestTan_lit.frag
    LLPC_SHADERTEST :: shaderdb/gfx9/ExtShaderFloat16_TestAngleTrigFuncs_lit.frag
I will investigate why.

Reply:
OK, so that one is needed too. One of the specific things that the TLI is needed for is so everyone knows we don't have tan or pow library functions.

Reply:
No, the only other caller of LgcContext::preparePassManager is SpirvLower::addPasses, which is preparing the per-shader pass manager not the per-pipeline one. And LgcContext::preparePassManager itself is still required, because if you don't create a target-specific TargetLibraryInfoWrapperPass then the pass manager creates a target-agnostic one for you.

Reply:
(My "No" was in response to your original question "Doesn't addTargetPasses in the back-end add it for us?".)